src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.TestClasses.cs

```diff
@@ -1067,5 +1068,25 @@ public new TestSettingsEnum2 E
             public override int X { set => base.X = value + 1; }
         }
 
+        public class EnumerableNotCollection : IEnumerable<KeyValuePair<string, string>>
```

Comment:
What specific attribute of this type makes it not a collection? Is it that it doesn't implement one of the known collection interfaces or that it's not mutable?

Reply:
It doesn't match any of the heuristics for enum-like collections: https://github.com/dotnet/runtime/blob/dcf9814b9194cd1ab14ed8108098f06e0fd53e67/src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs#L450-L540

In this particular case it's because it doesn't implement ICollection<>, nor expose an `Add` method: https://github.com/dotnet/runtime/blob/dcf9814b9194cd1ab14ed8108098f06e0fd53e67/src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs#L485